spring-cloud/spring-cloud-openfeign/src/main/java/com/baeldung/cloud/openfeign/exception/BadRequestException.java

```diff
@@ -0,0 +1,29 @@
+package com.baeldung.cloud.openfeign.exception;
+
+public class BadRequestException extends Exception{
+
+    private String message;
+
+    public BadRequestException() {
+        super();
+    }
+
+    public BadRequestException(String message) {
+        super();
+        this.message = message;
+    }
+
+    public BadRequestException(Throwable cause) {
+        super(cause);
+    }
+
+    @Override
+    public String toString() {
+        return "BadRequestException: "+message;
```

Comment:
Watch out for spacing around `+` - you could use `getMessage()` instead of this class's `message`